analyzers/src/SonarAnalyzer.CSharp/Rules/RedundancyInConstructorDestructorDeclaration.cs

```diff
@@ -48,11 +48,17 @@ protected override void Initialize(SonarAnalysisContext context)
         private static void CheckTypesWithPrimaryConstructor(SonarSyntaxNodeReportingContext context)
         {
             var typeDeclaration = (TypeDeclarationSyntax)context.Node;
-            if (typeDeclaration.ParameterList() is { Parameters.Count: 0 } parameterList
+            if (!Excluded(typeDeclaration)
+                && typeDeclaration.ParameterList() is { Parameters.Count: 0 } parameterList
                 && !IsStructWithInitializedFieldOrProperty(typeDeclaration, context.SemanticModel))
             {
                 context.ReportIssue(Rule, parameterList, "primary constructor");
             }
+
+            static bool Excluded(TypeDeclarationSyntax node) =>
+                RecordDeclarationSyntaxWrapper.IsInstance(node)
+                && ((RecordDeclarationSyntaxWrapper)node).BaseList is { } baseList
```

Comment:
In the Symbolic Execution engine we have `As...Operation` methods to simplify these kind of patterns. Do we not have something comparable for SyntaxWrappers?

Reply:
No sure - do we have something that works with structs?

Reply:
Probably not.

Reply:
Well I deleted this line anyways so this convo is not relevant anymore. All good.